.github/actions/lint/lint_links.py

```diff
@@ -76,13 +126,14 @@ def lint_markdown_files(files, pattern):
     return passed
 
 
-def check_markdown_file(filename, pattern):
+def check_markdown_file(filename, pattern, ignore_patterns):
     """
```

Comment:
This function is starting to get pretty dense. It would be prudent to abstract parts of this function to make for a simpler function. My rule of thumb is to have at most 2 levels of indentation for a given function, and we have up to 4 here. Consider making the following blocks into supporting helpers:

---
https://github.com/UBCSailbot/docs/blob/d2e726d4b34f17c0f1c913fc04c254616c1788b5/.github/actions/lint/lint_links.py#L148-L153
---
https://github.com/UBCSailbot/docs/blob/d2e726d4b34f17c0f1c913fc04c254616c1788b5/.github/actions/lint/lint_links.py#L154-L158
---